torchvision/models/detection/generalized_rcnn.py

```diff
@@ -77,6 +77,14 @@ def forward(self, images, targets=None):
             original_image_sizes.append((val[0], val[1]))
 
         images, targets = self.transform(images, targets)
+        if targets is not None:
+            for target in targets:
+                boxes = target["boxes"]
+                nonpostive_widths = (torch.sum(boxes[:, 2] <= boxes[:, 0]) > 0)
+                nonpostive_heights = (torch.sum(boxes[:, 3] <= boxes[:, 1]) > 0)
+                if nonpostive_widths > 0 or nonpostive_heights > 0:
+                    raise ValueError("Expected all bounding boxes to have positive height and width.")
```

Comment:
Can you maybe print the offending bounding box in the error message, so that the users can know more precisely what is going on? Maybe indicating the index in the `targets` as well.